src/compiler/crystal/codegen/codegen.cr

```diff
@@ -2036,7 +2036,7 @@ module Crystal
       if raise_overflow_fun = @raise_overflow_fun
         check_main_fun RAISE_OVERFLOW_NAME, raise_overflow_fun
       else
-        raise "BUG: __crystal_raise_overflow is not defined"
+        raise "ERROR: missing __crystal_raise_overflow function, either use std-lib's prelude or define it"
```

Comment:
```suggestion
        raise LocationlessException.new("Missing __crystal_raise_overflow function, either use std-lib's prelude or define it")
```

I think that should work fine. The report of exceptions should already include the Error: prefix.

@3n-k1 can you double-check?